### PR TITLE
Significantly improve download artifact latencies on GCE Windows node boot.

### PR DESCRIPTION
**Pull request by jeremyje**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
`MustDownload-File` uses `Invoke-WebRequest` which is slow for downloading files.
This change introduces the usage of `System.Net.Http.HttpClient` which is an updated version of `WebClient` to download these files.

We should observe a slight reduced boot time (mainly the Kubernetes binary download) in test runs.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/gce/windows/common.psm1 [hunk 1]
```diff
@@ -36,10 +36,13 @@ $ProgressPreference = 'SilentlyContinue'
...
 Export-ModuleMember -Variable REDO_STEPS
 
+# Add types used by the logic below
+Add-Type -AssemblyName System.Net.Http
+
 # Writes $Message to the console. Terminates the script if $Fatal is set.
 function Log-Output {
   param (
-    [parameter(Mandatory=$true)] [string]$Message,
+    [parameter(Mandatory = $true)] [string]$Message,
```
[the remark this answers is not in the document]
Comment (jeremyje) [reply cluster/gce/windows/common.psm1:45]: This was a consequence of saving the file in vscode.

File: cluster/gce/windows/common.psm1 [hunk 2]
```diff
@@ -128,43 +131,101 @@ function Validate-SHA {
...
+
+  $currentAttempt = 0
+  while ($true) {
+    foreach ($url in $URLs) {
+      if (($Attempts -ne 0) -And ($currentAttempt -Gt 5)) {
+        throw "Attempted to download ${url} ${currentAttempt} times. Giving up."
       }
+      $currentAttempt++
+      try {
+        # Use HttpClient in favor of WebClient.
```
[the remark this answers is not in the document]
Comment (jeremyje) [reply cluster/gce/windows/common.psm1]: New method `Get-RemoteFile` that will focus simply on downloading the file.

File: cluster/gce/windows/common.psm1 [hunk 3]
```diff
@@ -128,53 +128,132 @@ function Validate-SHA {
...
-      # Attempt to download the file
-      Try {
-        # TODO(mtaufen): When we finally get a Windows version that has Powershell 6
-        # installed we can set `-MaximumRetryCount 6 -RetryIntervalSec 10` to make this even more robust.
-        $result = Invoke-WebRequest $url -Headers $Headers -OutFile $OutFile -TimeoutSec 300
-      } Catch {
+  $currentAttempt = 0
+  while ($true) {
+    foreach ($url in $URLs) {
+      if (($Attempts -ne 0) -And ($currentAttempt -Gt 5)) {
```
Comment (pjh) [inline cluster/gce/windows/common.psm1:163]: Use lowercase for -and and -gt? This is what the style guide suggests, and would be consistent with the rest of this code I think.

File: cluster/gce/windows/common.psm1 [hunk 4]
```diff
@@ -128,53 +128,132 @@ function Validate-SHA {
...
+        throw "Attempted to download ${url} ${currentAttempt} times. Giving up."
+      }
+      $currentAttempt++
+      try {
+        Get-RemoteFile -OutFile $OutFile -Url $url -Headers $Headers
```
Comment (pjh) [inline cluster/gce/windows/common.psm1:168]: Thanks, this is much better!

**Review by pjh — APPROVED** [review]
One nit, if you don't feel like updating and re-sending the PR then it's fine.

**Pull request merged**